### DISCUSSION TITLE
Add support for `rye` as an environment manager

### DISCUSSION THREAD

**Discussion by marrov** [drivendataorg/cookiecutter-data-science#398] [Requested tools]
Hi all!

Just wanted to start by saying thanks to the maintainers and devs. This project is fantastic and we have been using it in our organization for some time. I also feel v2 has been a great improvement.

This discussion is to argue in favour of adding [`rye`](https://rye.astral.sh/guide/basics/) as a package manager. I've seen `ruff` is rightly being considered (discussion [here](https://github.com/drivendataorg/cookiecutter-data-science/discussions/374)) and `rye` is astral's take an a python package manager plus other goodies. It uses `ruff` and optionally `uv` under the hood (faster drop in replacement for `pip`) and acts like a better `poetry` (also requested [here](https://github.com/drivendataorg/cookiecutter-data-science/discussions/375)). We have been quite happy with it at my organization with it despite it being a relatively recent project. Actually, we have such a high conviction with `rye` that it is the main reason we are currently moving out of `cookiecutter-data-science` as our project templating solution.

It would be great if there could be support for `rye` within this project. I have used it quite extensively and would be happy to contribute with adding this tool with some guidance.

**Comment by pjbull** [comment]
Since astral took over rye, they've had the following note [on the docs and repo](https://github.com/astral-sh/rye).

> If you're getting started with Rye, consider [uv](https://github.com/astral-sh/uv), the [successor project](https://lucumr.pocoo.org/2024/2/15/rye-grows-with-uv/) from the same maintainers.
> 
> While Rye is actively maintained, uv offers a more stable and feature-complete experience, and is the recommended choice for new projects.
> 
> Having trouble migrating? https://github.com/astral-sh/rye/discussions/1342


Given that, I think we're not likely to add rye support since it's likely just to get less and less attention as uv is developed.

**Reply by WeiXie03** [reply]
Hello there,

Firstly, I want to echo @marrov's words of thanks and wonderful experience using cookiecutter-data-science.

I actually started off using `uv` and only learned of `rye` after reading this original post.
In this case, would the team be considering supporting`uv` as a package manager for `cookiecutter-data-science`?

Thanks!

**Reply by pjbull** [reply]
Thanks @WeiXie03! We already support `uv` in the latest version https://cookiecutter-data-science.drivendata.org/all-options/#environment-manager
